src/librustc_typeck/astconv.rs

```diff
@@ -306,6 +306,21 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             };
         }
 
+        for (param, binding) in def.params.iter().zip(args.bindings.iter()) {
+            match param.kind {
+                GenericParamDefKind::Lifetime => {}
+                GenericParamDefKind::Type { has_default, .. } => {
+                    if has_default {
+                        tcx.check_stability(param.def_id, Some(binding.hir_id), binding.span);
```

Comment:
I don't believe this is getting run. I'm slowly debugging, sadly my laptop takes ages to compile.

Reply:
I suspect I should have been zipping with `args.args`. In a few hours when it finishes compiling again I'll know.

Reply:
@varkor When you say
> However, if you get `encountered unmarked API errors`, you'll need to explicitly mark them as stable in visit_generic_param

Is this what you mean? 
I'm still getting `encountered unmarked API errors` even if I put it at the top of `visit_generic_param`, so I suspect I have made a mistake.
```rust
let feature = Symbol::intern("Generic parameters default to stable");
let stab = self.tcx.intern_stability(Stability {
    level: attr::StabilityLevel::Stable {
    since: feature,
    },
    feature,
    rustc_depr: None,
    const_stability: None,
    promotable: false,
    allow_const_fn_ptr: false,
});
```

Reply:
I think you shouldn't need to do anything explicitly to mark it as stable, so I think something else is going wrong here.

Reply:
It does not occur when `tcx.check_stability(param.def_id, Some(arg.id()), arg.span())`  is omitted. Any thoughts on what might cause such a thing?

```rust
error: internal compiler error: encountered unmarked API: DefId(11:1404 ~ hashbrown[6e25]::map[0]::HashMap[0]::S[0])
   --> src/libstd/collections/hash/map.rs:203:31
    |
203 |     base: base::HashMap<K, V, S>,
    |                               ^

```